core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -182,10 +189,8 @@ void forceRefresh() {
    * would expire.
    */
   private InstanceData performRefresh() throws InterruptedException, ExecutionException {
-    logger.fine("Refresh Operation: Acquiring rate limiter permit.");
-    // To avoid unreasonable SQL Admin API usage, use a rate limit to throttle our usage.
-    forcedRenewRateLimiter.acquirePermit();
-    logger.fine("Refresh Operation: Acquired rate limiter permit. Starting refresh...");
+    logger.fine("Refresh Operation: Attempting refresh.");
+    rateLimitCalculator.recordAttempt();
```

Comment:
We handle rate limits in the delay until next refresh, instead of blocking the thread.